components/UserObserver.php

```diff
@@ -17,46 +17,53 @@
  */
 
-class UserObserver
+//namespace OEModule\CSDClient\components;
+
+use OEModule\mehstaffdb\components\CSDClient\CSDClient;
+
+class UserObserver extends \BaseAPI
 {
-	public function updateUser($params)
-	{
-		if (in_array($params['username'],Yii::app()->params['local_users'])) {
+
+	/**
+     * @return CSDClient
+     */
+	protected function getCSDClient(): CSDClient
+    {
+        return CSDClient::get();
+    }
+
+
+	/**
+     * Updates user from CSD database
+     *
+     * @param array $params
+	 * @return User
+     */
+	public function updateUser($params) :User
+	{		
+		/*if (in_array($params['username'],Yii::app()->params['local_users'])) {
 			return;
-		}
+		}*/
+
+		//if (Yii::app()->params['mehstaffdb_always_refresh'] || $this->isStale($params['username'])) {
+			//try {
+				$username = "WILLIAMSS";//$params['username'];
+				$remote_user = $this->getCSDClient()->getUserData($username);
+				if ($remote_user = $this->getCSDClient()->getUserData($username)) {
+					$remote_user = json_decode($remote_user, true);
+
+					$user = $this->getUser($username);
+					$user_authentication = $this->getUserAuthentication($username);
 
-		if (Yii::app()->params['mehstaffdb_always_refresh'] || $this->is_stale($params['username'])) {
-			try {
-				if ($remote_user = StaffDB_User::model()->find("MUUID_Staff_DomainUsername=?", array($params['username']))) {
-					if (!$user = User::model()->find('username=?',array($params['username']))) {
+					if (!$user) {
 						$user = new User;
 						$preexists = false;
 					} else {
 						$preexists = true;
 					}
 
-					$user->code = $remote_user->MUUID_Staff_MUUID;
-					$user->username = $remote_user->MUUID_Staff_DomainUsername;
-					$user->first_name = $remote_user->MUUID_Staff_NameFirst;
-					$user->last_name = $remote_user->MUUID_Staff_NameLast;
-					$user->email = Yii::app()->params['mehstaffdb_default_email'];
-					$user->title = $remote_user->MUUID_Staff_Title;
-					$user->qualifications = $remote_user->EPR_MedicalDegrees;
-					$user->role = $remote_user->MUUID_Staff_JobTitle;
-					$user->doctor_grade_id = $this->getDoctorGradeFromJobTitle($remote_user->MUUID_Staff_JobTitle);
-					$user->registration_code = $this->getGMCRegistrationNumber($remote_user->MUUID_Staff_PersonnelID);
-					$user->password = 'faed6633f5a86241f3e0c2bb2bb768fd';
-					$user->is_doctor = $remote_user->MUUID_Staff_IsDoctor;
-					$user->is_clinical = $remote_user->MUUID_Staff_IsClinical;
-					$user->is_consultant = $remote_user->MUUID_Staff_IsConsultant;
-					$user->is_surgeon = $remote_user->MUUID_Staff_IsSurgeon;
-					$user->active = !$remote_user->MUUID_Staff_LeftMEH;
-					$user->global_firm_rights = 1;
-
-					if (!$user->save(false)) {
-						throw new Exception('Unable to save user: '.print_r($user->getErrors(),true));
-					}
-
+					$user = $this->saveUser($user, $user_authentication, $remote_user);
+					
 					if (!$preexists) {
 						$contact = new Contact;
```

Comment:
similar as above, should it be `$contact = new Contact();` ?

Reply:
same as below, inside of else

Reply:
Changed code for `$contact = new Contact();`